app/models/user.rb

```diff
@@ -1,17 +1,43 @@
+# frozen_string_literal: true
+
+# user model
 class User < ApplicationRecord
-    before_save { self.email = email.downcase }
-    validates :name,  presence: true, length: { maximum: 50 }
-    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
-    validates :email, presence: true, length: { maximum: 255 },
-                      format: { with: VALID_EMAIL_REGEX },
-                      uniqueness: { case_sensitive: false }
-    has_secure_password
-    validates :password, presence: true, length: { minimum: 6 }
+  attr_accessor :remember_token
+  before_save { self.email = email.downcase }
+  validates :name,  presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6 }
+
+  # Returns the hash digest of the given string.
+  def self.digest(string)
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+                                                BCrypt::Engine.cost
+    BCrypt::Password.create(string, cost: cost)
+  end
+
+  # Returns a random token.
+  def self.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
```

Comment:
@kusanagi98 
Dùng update_attributes

Reply:
"Meanwhile, the second line of remember uses the **update_attribute** method to update the remember digest. (As noted in Section 6.1.5, this method bypasses the validations, which is necessary in this case because **we don’t have access to the user’s password or confirmation**.)" sách nó bảo thế này a ơi :v

Reply:
@kusanagi98 
https://stackoverflow.com/questions/7083575/updating-user-attributes-without-requiring-password
Có thể dùng cách như trên để skip validation chỉ cho password. Nếu skip tất cả các validation thì dễ gặp trường hợp sinh ra bug sau này

Reply:
ok r a ơi